models/software_version_helpers.go

```diff
@@ -0,0 +1,176 @@
+	switch {
+	case err == nil:
+		return nil, errors.Errorf("service software versions with id '%s' already exists", id)
+	case errors.Is(err, ErrNotFound):
+	default:
+		return nil, errors.WithStack(err)
+	}
+
+	row := &ServiceSoftwareVersions{
+		ID:               id,
+		ServiceID:        params.ServiceID,
+		SoftwareVersions: params.Versions,
+		CheckAt:          params.CheckAt,
+	}
+
+	if err := q.Insert(row); err != nil {
+		return nil, errors.Wrap(err, "failed to insert service software versions")
+	}
+
+	return row, nil
+}
+
+// UpdateServiceSoftwareVersionsParams represents params for updating service software versions entity.
+type UpdateServiceSoftwareVersionsParams struct {
+	CheckAt  *time.Time
+	Versions *[]SoftwareVersion
+}
+
+// UpdateServiceSoftwareVersions updates existing service software versions.
+func UpdateServiceSoftwareVersions(
+	q *reform.Querier,
+	id string,
+	params UpdateServiceSoftwareVersionsParams,
+) (*ServiceSoftwareVersions, error) {
+	row, err := FindServiceSoftwareVersions(q, id)
+	if err != nil {
+		return nil, err
+	}
+	if params.CheckAt != nil {
+		row.CheckAt = *params.CheckAt
+	}
+	if params.Versions != nil {
+		row.SoftwareVersions = *params.Versions
+	}
+
+	if err := q.Update(row); err != nil {
+		return nil, errors.Wrap(err, "failed to update service software versions")
+	}
+
+	return row, nil
+}
+
+// FindServiceSoftwareVersions returns service software versions entry by given ID if found, ErrNotFound if not.
+func FindServiceSoftwareVersions(q *reform.Querier, id string) (*ServiceSoftwareVersions, error) {
+	if id == "" {
+		return nil, errors.New("service software versions id is empty")
+	}
+
+	versions := &ServiceSoftwareVersions{ID: id}
+	switch err := q.Reload(versions); err {
+	case nil:
+		return versions, nil
+	case reform.ErrNoRows:
+		return nil, errors.Wrapf(ErrNotFound, "service software versions by id '%s'", id)
+	default:
+		return nil, errors.WithStack(err)
+	}
+}
+
+// FindServiceSoftwareVersionsByServiceID returns service software versions entry
+// by given service ID if found, ErrNotFound if not.
+func FindServiceSoftwareVersionsByServiceID(q *reform.Querier, serviceID string) (*ServiceSoftwareVersions, error) {
+	if serviceID == "" {
+		return nil, errors.New("service id is empty")
+	}
+
+	s, err := q.SelectOneFrom(ServiceTable, "WHERE service_id = $1", serviceID)
+	switch err {
+	case nil:
+		return s.(*ServiceSoftwareVersions), nil
+	case reform.ErrNoRows:
+		return nil, errors.Wrapf(ErrNotFound, "service software versions by service id '%s'", serviceID)
+	default:
+		return nil, errors.WithStack(err)
+	}
+}
+
+// FindServicesSoftwareVersions returns services software versions.
+func FindServicesSoftwareVersions(q *reform.Querier, limit *int) ([]*ServiceSoftwareVersions, error) {
+	var args []interface{}
+	var limitStatement string
+	if limit != nil {
+		limitStatement = " LIMIT $1"
+		args = append(args, *limit)
+	}
+
+	const orderByField = "check_at"
```

Comment:
Does it make sense to have it as const? I mean why not just `fmt.Sprintf("ORDER BY check_at %s"...`?

Reply:
there was additional ordering by ID. Constant removed.